File: go/aprenda-go-youtube/fatorial.go
package main

import "fmt"

func main() {
	fmt.Println("calculo do fatorial")
	x := calcular(50)
	fmt.Println(x)

}

func calcular(x int) int {
	if x == 1 {
		return 1
	}
	return x * calcular(x-1)
}

File: elixir/alura/fatorial.exs
defmodule Fatorial do
  # calcular o fatorial:  n! = n . (n – 1). (n – 2). (n – 3)
  def fatorial(0), do: 1

  def fatorial(numero) when is_integer(numero) and numero > 0 do
    numero * fatorial(numero - 1)
  end

  def fatorial(_), do: IO.puts("informe numero positivo")
end
